### PR TITLE
3Sum done, took ages to test

### PR DESCRIPTION
**Pull request by loresfca**
Had some trouble uploading the TEBOW IT picture, I will try to add it from another computer as soon as possible.

### CODE REVIEW

File: Week 3/Solution.java [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+import org.junit.Test;
+import static org.junit.Assert.assertEquals;
+public class Solution {
+    public List<List<Integer>> threeSum(int[] n) {
+        List<List<Integer>> res = new  ArrayList<List<Integer>>();
+        if(n == null || n.length<3) return res;
+        int len = n.length;
+        HashSet<Integer> checked = new HashSet<Integer>();
+        for(int i = 0; i<len; i++){
+            int goal = n[i]*-1;
```
Comment (maknoon) [inline Week 3/Solution.java:12]: this was how I ended up solving it too, yay! one suggestion I have is putting some more comments in your code. also, good idea to use `HashSet<Int>` to keep track of duplicates in the input array. 👍

Comment (loresfca) [reply Week 3/Solution.java:12]: Thanks! I'll try to add more comments 😄

File: Week 3/TEBOW.txt [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+Input: [1,2,1,-2] 
+Expected:[[-2,1,1]]
+-Array with duplicates and multiple answers
+Input: [1,2,1,-2,-3] 
+Expected: [[-2,1,1],[-3,1,2]]
+-Array where there would be an answer if the element were duplicated but it is not
+Input: [1,2,-2] Expected:[]
+-Array where the answer is 0
+Input: [0,0,0] Expected: [[0,0,0]]
+Brute Force:
```
Comment (charientism) [inline Week 3/TEBOW.txt]: Definitely a solid brute force - thanks for including the Big O, that's good practice :)

File: Week 3/TEBOW.txt [hunk 3]
```diff
@@ -0,0 +1,64 @@
+Given an array S of n integers, are there elements a, b, c in S such that a + b + c = 0? Find all unique triplets in the array which gives the sum of zero.
+
+Talk:
```
Comment (charientism) [inline Week 3/TEBOW.txt]: Awesome TEBOW IT! Thanks for taking the time to fill it out :)

File: Week 3/Solution.java [hunk 4]
```diff
@@ -0,0 +1,64 @@
+import java.io.*;
```
Comment (charientism) [inline Week 3/Solution.java:1]: Good job with this question - making use of the 2Sum algorithm is definitely a great way to go

File: Week 3/Solution.java [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+        newTriplet.add(one);
+        newTriplet.add(two);
+        newTriplet.add(three);
+        Collections.sort(newTriplet);
+        if(!res.contains(newTriplet)){
+            res.add(newTriplet);
+        }       
+    }
+    @Test
+   public void testThreeSum() {
```
Comment (charientism) [inline Week 3/Solution.java:38]: Solid testing. Good job picking equivalence classes and keeping your input small. The easier the examples are to step through, the better.

**Review by charientism — APPROVED** [review]

**Pull request merged**